File: back-end/models/Administrator.go
package models

import "time"

// AccessType struct
type AccessType int

const (
	Admin AccessType = iota // 0
)

func (a AccessType) String() string {
	return [...]string{"Admin", "Manager"}[a]
}

// Administrator struct
type Administrator struct {
	IdAdministrator uint       `gorm:"primaryKey" json:"id_administrator"`
	Name            string     `json:"name"`
	Email           string     `json:"email"`
	Nickname        string     `json:"nickname"`
	Password        string     `json:"password"`
	AccessType      AccessType `json:"access_type"`
	IsActive        bool       `json:"is_active"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

File: front-end/web/src/app/models/administrator.ts
export class Administrator {
  id_administrator: number;
  name: string;
  email: string;
  nickname: string;
  password: string;
  access_type: number;
  is_active: boolean;
  created_at: Date;

  constructor() {
    this.id_administrator = 0;
    this.name = '';
    this.email = '';
    this.nickname = '';
    this.password = '';
    this.access_type = 0;
    this.is_active = true;
    this.created_at = new Date();
  }
}
